compiler/frontend/beam/src/main/java/org/apache/nemo/compiler/frontend/beam/transform/GroupByKeyAndWindowDoFnTransform.java

```diff
@@ -171,35 +227,73 @@ public String toString() {
     return timerData;
   }
 
+  /**
+   * Maintains state internals and timer internals for keys.
+   */
+  final class StateAndTimerInternalsFactory {
+    private final InMemoryTimerInternalsFactory inMemoryTimerInternalsFactory;
```

Comment:
Can this and the below variables be simply fields of `GroupByKeyAndWindowDoFnTransform`?

Reply:
Done